### PR TITLE
Feat(#31): Add ignore option

### PR DESCRIPTION
**Pull request by hinok**
- This commit add functionality to ignore files
based on provided regexps.
- in index.js config must be specified as below

    // @create-index {"ignore":["/foo.js"]}

### CODE REVIEW

File: src/utilities/log.js [hunk 1]
```diff
@@ -2,8 +2,6 @@ import chalk from 'chalk';
 import moment from 'moment';
 
 export default (...append) => {
-    /* eslint-disable no-console */
+  /* eslint-disable no-console */
```
Comment (gajus) [inline src/utilities/log.js]: Use `// eslint-disable-next-line`

File: src/utilities/constants.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const CREATE_INDEX_PATTERN = /(?:^|[\n\r]+)\/\/ @create-index\s?({.*})?[\n\r]+/;
```
Comment (gajus) [inline src/utilities/constants.js]: `export const CREATE_INDEX_PATTERN ...`

File: package.json [hunk 3]
```diff
@@ -48,7 +48,8 @@
     "create-index": "node ./dist/bin/create-index ./src/utilities",
     "lint": "cross-env NODE_ENV=development eslint ./src ./tests",
     "precommit": "npm run test",
-    "test": "npm run build && npm run lint && cross-env NODE_ENV=development mocha --compilers js:babel-register"
+    "test": "npm run build && npm run lint && cross-env NODE_ENV=development mocha --compilers js:babel-register",
+    "test:watch": "cross-env NODE_ENV=development mocha --watch --reporter spec --bail --compilers js:babel-register"
```
Comment (gajus) [inline package.json]: Remove `test:watch`.

**Comment by gajus** [comment]
Thank you for the PR.

This lack documentation, `README.md` changes.

File: test/writeIndex.js [hunk 4]
```diff
@@ -8,29 +8,59 @@ import {
...
-    }
+export { default as bar } from './bar';
+export { default as foo } from './foo.js';
+    `));
+  });
 
-    writeIndex([path.resolve(fixturesPath, 'mixed')]);
+  it('creates index with config in target directory', () => {
+    const indexFilePath = path.resolve(fixturesPath, 'with-config/index.js');
+    const ignoredExportLine = `export { default as bar } from './bar.js';`; // eslint-disable-line quotes
```
Comment (gajus) [inline test/writeIndex.js]: Use `// eslint-disable-next-line`

File: test/writeIndex.js [hunk 5]
```diff
@@ -8,29 +8,59 @@ import {
...
+  // eslint-disable-next-line no-empty
+  try {
+    fs.unlinkSync(filePath);
+  } catch (error) {
+
+  }
+};
+
+const appendToFile = (filePath, content) => {
+  try {
```
Comment (gajus) [inline test/writeIndex.js]: Whats the reason for swallowing an error?

File: test/writeIndex.js [hunk 6]
```diff
@@ -8,29 +8,59 @@ import {
 import writeIndex from '../src/utilities/writeIndex';
 import codeExample from './codeExample';
 
+const readFileByPath = (filePath) => {
+  return fs.readFileSync(filePath, 'utf8');
+};
+
+const removeFileByPath = (filePath) => {
+  // eslint-disable-next-line no-empty
+  try {
```
Comment (gajus) [inline test/writeIndex.js]: Whats the reason for swallowing an error?

File: src/utilities/readIndexConfig.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import fs from 'fs';
+import path from 'path';
+import hasIndex from './hasIndex';
+import {CREATE_INDEX_PATTERN} from './constants';
+
+const isString = (str) => {
```
Comment (gajus) [inline src/utilities/readIndexConfig.js]: Remove unnecessary helpers. I am all pro declarative programming, though `typeof str === 'string'` is already achieving the same.

File: src/utilities/readIndexConfig.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  return typeof str === 'string';
+};
+
+const isEmptyString = (str) => {
+  return str.length === 0;
```
Comment (gajus) [inline src/utilities/readIndexConfig.js]: Remove unnecessary helpers. I am all pro declarative programming, though `str.length` is already achieving the same.

File: test/writeIndex.js (hunk 6 above)
Comment (hinok) [inline test/writeIndex.js]: I was trying to follow "code style" in the project so if you look here https://github.com/gajus/create-index/pull/37/files#diff-f546cc5537a07f2741198b32e7ef3aa9L17 the same pattern with swallowing an error has been already used. I decided to just copy and use it.

Anyway I will remove them if you want, no problem :)

Comment (gajus) [inline test/writeIndex.js]: Oh, thats in the `test/`. Meh, I am indifferent.

**Comment by hinok** [comment]
Sorry for so many changes. I applied all requested changes and removed "swallowing errors" from tests. I added also section about `ignore` in `README.md`.

**Comment by gajus** [comment]
Is this ready to be merged?

**Comment by hinok** [comment]
I thought that `writeIndexCli.js` uses `writeIndex.js`, it doesn't - my fault, I didn't check but I fixed it already.

I also checked this PR on my local project where I use `create-index` heavily to maintain icons, it works 💃 

Please feel free to correct any english language issues.
I think that it's ready to be merged.

**Comment by gajus** [comment]
Thank you

**Pull request merged**